src/databricksbundle/notebook/GithubLinkGenerator.py

```diff
@@ -0,0 +1,90 @@
+import tomlkit
+import traceback
+
+from inspect import getsource
+from pathlib import Path
+
+
+class GithubLinkGenerator:
+    def __get_repo_name(self, module: str) -> str:
+        special_repos = {"daipecore": "daipe-core", "featurestorebundle": "feature-store-bundle"}
+
+        if module in special_repos:
+            return special_repos[module]
+
+        return self.__get_repo_name_with_bundle(module)
+
+    def __get_repo_name_with_bundle(self, module: str) -> str:
+        index_of_bundle = module.find("bundle")
+
+        return module[:index_of_bundle] + "-" + module[index_of_bundle:]
+
+    def __get_module_version(self, module_name: str) -> str:
+        base_path = Path("/".join(list(Path.cwd().parts[0:5])))
+        lockfile_path = base_path.joinpath("poetry.lock")
+
+        with lockfile_path.open("r") as f:
+            config = tomlkit.parse(f.read())
+
+        for module in config["package"]:
+            if module["name"] == module_name:
+                return module["version"]
+
+        raise Exception("Specified module could not be found, make sure the the module is imported and developed by DAIPE")
+
+    def __get_display_html(self):
+        import IPython
+
+        ipython = IPython.get_ipython()
+
+        if not hasattr(ipython, "user_ns") or "displayHTML" not in ipython.user_ns:
+            raise Exception("displayHTML cannot be resolved")
+
+        return ipython.user_ns["displayHTML"]
+
+    def __get_github_url_from_module(self, module: object) -> str:
```

Comment:
I think the `module` is of type `type` try running `type()` on some class
<img width="293" alt="Screenshot 2022-01-20 at 13 36 25" src="https://user-images.githubusercontent.com/82815273/150339945-9415c27f-cb94-4757-a73f-5f9528cc95a3.png">
or it could be a type function, so maybe `Union[type, Callable]`?